src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2427,9 +2432,9 @@ void inferDeclarationExpressionValEscape()
                         : GetValEscape(fromArg, scopeOfTheContainingExpression));
                 }
 
-                foreach (var (_, fromArg, _, _) in escapeValues)
+                foreach (var argument in argsOpt)
```

Comment:
Since out arguments are usually not included in `escapeValues`, we needed to change to check `argsOpt` instead. Note that `escapeValues` was probably never the right collection to enumerate, since the same argument can be present multiple times, it just happened to work coincidentally until now.

Reply:
Also, although it doesn't look like 'argsOpt' is default-checked on this code path, I actually think that the argument is never default. I'll send a separate PR after this one to add an assertion and rename the parameter.